berkdb/lock/lock.c

```diff
@@ -2605,6 +2654,17 @@ __lock_get_internal_int(lt, locker, in_locker, flags, obj, lock_mode, timeout,
 			++sh_obj->generation;
 			unlock_detector(region);
 		}
+
+		extern u_int32_t gbl_rep_lockid;
+		if (locker == gbl_rep_lockid) {
```

Comment:
Can we optimize and do this only if action is  HEAD,TAIL, or SECOND?

Reply:
Also, can we just copy newl  and do the __lock_to_dbt() conversion when we need to dump the holder's list?

Reply:
> Can we optimize and do this only if action is HEAD,TAIL, or SECOND?

That covers all the cases.

Reply:
Sorry, I failed to see that we were already inside the outer switch. Ignore the first comment.